### PR TITLE
Print readable error message when check critical cfg

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**

### CODE REVIEW

**Comment by siddontang** [comment]
Em, I prefer returning error ASAP and then panic plus the message `remove last_tikv.toml`.

**Comment by hicqu** [comment]
I prefer print all changed critical configration at one time. Otherwise if user only fix the first and remove `last_tikv.toml`, we can't check any more critical changes.

File: src/config.rs [hunk 1]
```diff
@@ -773,38 +773,71 @@ impl TiKvConfig {
...
-            return Err(format!(
-                "storage data dir have been changed, former data dir is {}, \
-                 current data dir is {}, please check if it is expected.",
-                last_cfg.storage.data_dir, self.storage.data_dir
-            ).into());
-        }
+        if data_dir_changed || data_wal_dir_changed || raft_dir_changed || raft_wal_dir_changed {
+            let mut error_msg = String::new();
+            if data_dir_changed {
+                error_msg.push_str(
```
[the remark this answers is not in the document]
Comment (huachaohuang) [reply src/config.rs]: We can collect the diffs together, print them at once and tell the user to check them.

**Comment by breeswish** [comment]
LGTM

**Comment by BusyJay** [comment]
Any updates?

**Pull request closed without merging**